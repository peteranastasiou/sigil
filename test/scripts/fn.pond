fn a() {
    print("hello");
}

fn b(a, b) {
    return a + b;
}

fn c(x) {
    if x < 0 {
        return "A1";
    }
    return if x < 1 { "B1" } else { "B2" };
}

a();

print(b(1, 3/2));

print(c(-1));
print(c(0.5));
print(c(1.5));

# --------------------------------
# Alternate function syntax:
# --------------------------------

var d1 = fn(i) { return i*2; };
var d2 = fn(i) { i*2 };
fn d3(i) { i*2 }

print(d1(2));
print(d2(2));
print(d3(2));

var e1 = fn(i) { var g = i; return g*2; };
var e2 = fn(i) { var g = i; g*2 };
fn e3(i) { var g = i; g*2 }

print(e1(2));
print(e2(2));
print(e3(2));

fn function_runner(f) {
    if f() {
        echo "true";
    }else{
        echo "false";
    }
}

function_runner( fn() { return true; } );

# fn(i) { print("hello"); print(i); }(7);
