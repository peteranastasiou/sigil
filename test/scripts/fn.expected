hello
2.5
A1
B1
B2
4
4
4
4
4
4
"true"
hello
7
8
